docs/Addons.md

```diff
@@ -32,6 +30,17 @@ env:
   value: system:serviceaccount:%s:cluster-admin
 ```
 
+### Addon Cleanup
+
+If your addon test creates or affects anything outside of the OSD cluster lifecycle, a separate cleanup action is required. If `ADDON_RUN_CLEANUP` is set to `true`, OSDe2e will run the test harness a **second time** passing the argument `cleanup` to the container.
+
+There may be a case where a separate cleanup container/harness is required. That may be configured using the `ADDON_CLEANUP_HARNESSES` config option. It is formatted in the same way as `ADDON_TEST_HARNESSES`. This however, may cause some confusion as to what is run when:
+
+`ADDON_RUN_CLEANUP` is true, and `ADDON_CLEANUP_HARNESSES` is not set, OSDe2e will only run `ADDON_TEST_HARNESSES` again, passing the `cleanup` argument.
+
+`ADDON_RUN_CLEANUP` is true, and `ADDON_CLEANUP_HARNESSES` is set, OSDe2e will only run the `ADDON_CLEANUP_HARNESSES`, passing the `cleanup` argument.
```

Comment:
As the `ADDON_CLEANUP_HARNESSES` can only be run when the `ADDON_RUN_CLEANUP` flag is set, I don't see the need to pass in the `cleanup` argument.

Reply:
That's a fair point. :) I'll update the docs and logic quick.